lib/galaxy/datatypes/binary.py

```diff
@@ -1900,28 +1906,43 @@ def set_meta(self, dataset, overwrite=True, metadata_tmp_files_dir=None, **kwd):
 
     def sniff(self, filename):
         if super().sniff(filename):
-            keys = ["-model_config-"]
+            keys = [self._CONFIG]
             with h5py.File(filename, "r") as handle:
                 if not all(name in handle.keys() for name in keys):
                     return False
-                url = util.unicodify(handle.attrs.get("-URL-"))
+                url = util.unicodify(handle.attrs.get(self._URL))
             if url == self.URL:
                 return True
         return False
 
     def get_repr(self, filename):
         try:
             with h5py.File(filename, "r") as handle:
-                repr_ = util.unicodify(handle.attrs.get("-repr-"))
-            return repr_
+                repr_ = util.unicodify(handle.attrs.get(self._REPR))
+            if len(repr_) <= self.max_preview_size:
+                return repr_
+            else:
+                return "<p><strong>The model representation is too big to be displayed!</strong></p>"
         except Exception as e:
             log.warning("%s, get_repr Except: %s", self, e)
             return ""
 
+    def get_html_repr(self, filename):
```

Comment:
If you don't mind, could you factor out the common logic from these 2 methods into a helper method? That way you could get rid of 10 lines of (almost) duplicate code. Something like this, maybe:
```
def get_repr:
    return repr_helper(repr args)

def get_html_repr:
    return repr_helper(html-repr args)

def repr_helper(args):
    main logic, parametrized with args
```

Reply:
Nice point! How about now?

Reply:
Looks good, thank you!